exchange.c

```diff
@@ -1,51 +1,76 @@
+#include <gnu/libc-version.h>
+#include <linux/fs.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdbool.h>
+#include <sys/syscall.h>
+#include <unistd.h>
 
-void printHelp() {
-    puts("exchange: exchange names of two files or directories atomically\n"
-         "Usage: exchange FILE1 FILE2 [flags...]\n"
-         "Allowed flags:\n"
-         "\t--help:    shows this help text\n"
-         "\t--verbose: produces verbose output");
-}
+static struct option long_options[] = {
+    {"help",    no_argument, NULL, 'h'},
+    {"verbose", no_argument, NULL, 'v'},
+    {0, 0, 0, 0}
+};
 
-int main(int argc, char** argv) {
-    if (argc < 3) {
-        printHelp();
-        return 1;
-    }
+void usage(int exit_code) {
+    puts("Usage: exchange [-v] PATH1 PATH2\n"
+         "  or:  exchange -h|--help\n"
+         "atomically exchange names and contents of two files or directories\n"
+         "\n"
+         "Options:\n"
+         "  -h, --help      shows help text\n"
+         "  -v, --verbose   produce verbose output");
+    exit(exit_code);
+}
 
+int main(int argc, char **argv) {
+    int c;
     bool verbose = false;
-    for (int i = 3; i < argc; i++) {
-        if (!strcmp(argv[i], "--verbose")) {
-            verbose = true;
-        } else if (!strcmp(argv[i], "--help")) {
-            printHelp();
-            return 0;
-        } else {
-            printf("Unrecognized flag: %s", argv[i]);
-            return 1;
+    char **paths;
+    int ret;
+
+    while ((c = getopt_long(argc, argv, "hv", long_options, NULL)) != -1) {
+        switch (c) {
+            case 'h':
+                usage(0);
+            case 'v':
+                verbose = true;
+                break;
+            default:
+                usage(1);
         }
     }
 
+    if (argc - optind != 2) {
+        puts("Exactly two paths are required.\n");
+        usage(1);
+    }
+
+    paths = argv + optind;
+
     if (verbose) {
-        printf("Exchanging names: %s and %s\n", argv[1], argv[2]);
+        printf("Exchanging paths %s and %s\n", paths[0], paths[1]);
     }
 
-    // atomically exchange the two directories
-    // if paths are not absolute, consider them relative to the current working directory
-    if(renameat2(AT_FDCWD, argv[1], AT_FDCWD, argv[2], RENAME_EXCHANGE) == -1) {
+    // Atomically exchange the two paths. If paths are not absolute, consider
+    // them relative to the current working directory
+#if (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 28))
+        ret = renameat2(AT_FDCWD, paths[0], AT_FDCWD, paths[1], RENAME_EXCHANGE);
+#else
+        ret = syscall(SYS_renameat2, AT_FDCWD, paths[0], AT_FDCWD, paths[1], RENAME_EXCHANGE);
+#endif
+
+    if (ret == -1) {
         perror("exchange: could not exchange the two paths");
-        return 1;
+        exit(1);
     }
 
     if (verbose) {
-        puts("Names exchanged successfully");
+        puts("Paths exchanged successfully");
     }
-
-    return 0;
```

Comment:
I would suggest keeping the `return 0` here.

Reply:
`return 0` is implicit, but that's fine. I went a step further and replaced 0 and 1 exit codes with EXIT_SUCCESS and EXIT_FAILURE from stdlib.h.